src/v/http/client.cc

```diff
@@ -589,4 +590,25 @@ ss::input_stream<char> client::response_stream::as_input_stream() {
     return ss::input_stream<char>(std::move(ds));
 }
 
+client::request_header redacted_header(client::request_header original) {
+    using field_type = std::variant<boost::beast::http::field, std::string>;
+
+    static const std::unordered_set<field_type> redacted_fields{
+      boost::beast::http::field::authorization, "x-amz-content-sha256"};
+
+    auto h{std::move(original)};
+
+    for (const auto& field : redacted_fields) {
+        std::visit(
+          [&h](const auto& f) {
+              if (h.find(f) != h.end()) {
+                  h.set(f, "*****");
+              }
+          },
+          field);
+    }
```

Comment:
nit: we use `O( size of redacted_fields * size of original)` operations here, but if we iterate over `h` and will call `redacted_fields.find()`, then we will use `O(size of original)` since `unordered_set::find` has constant amortized complexy.

Reply:
If I understand correctly, right now we use `O(redacted_fields.size() * log(original.size())` because `h.find` is run on a [multiset internally](https://github.com/boostorg/beast/blob/develop/include/boost/beast/http/impl/fields.hpp#L710) which is supposed to have logarithmic search. 

If we iterate over `original` then we will increase the runtime to `O(original.size())`.

Note: If `h.find()` succeeds an `iterator_to` method is called but it also apparently [constant time](https://www.boost.org/doc/libs/1_79_0/doc/html/boost/intrusive/list.html#idm34142-bb)

Reply:
makes sense, thanks!